crates/turbopack-ecmascript/src/chunk/placeable.rs

```diff
@@ -36,8 +39,19 @@ enum SideEffectsValue {
 #[turbo_tasks::function]
 async fn side_effects_from_package_json(
     package_json: Vc<FileSystemPath>,
+    side_effect_free_packages: Vc<Vec<String>>,
 ) -> Result<Vc<SideEffectsValue>> {
     if let FileJsonContent::Content(content) = &*package_json.read_json().await? {
+        if let Some(pacakge_name) = content.get("name").and_then(|v| v.as_str()) {
+            if side_effect_free_packages
+                .await?
+                .iter()
+                .any(|v| pacakge_name == v)
+            {
+                return Ok(SideEffectsValue::Constant(false).cell());
+            }
+        }
```

Comment:
I don't feel good about the `package.json` name field for that. That's also not what next.js is using for `optimizePackageImports`. I think it should be a glob on the file path. You can apply that in `fn is_marked_as_side_effect_free`